File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let queue = [];

    let days = progresses.map((e , i) => Math.ceil((100 - e) / speeds[i]));
    let max = days[0];
    let count = 0;
    
    for(let i = 0; i < progresses.length; i++){
        if(days[i] <= max) count++;
        else{
            answer.push(count);
            count = 1;
            max = days[i];
        }
    }
    answer.push(count);
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    
    answer = []
    progress_queue = deque(progresses)
    speeds_queue = deque(speeds)
    
    while len(progress_queue) > 0:
        for i in range(len(progress_queue)):
            progress_queue[i] += speeds_queue[i]
            if progress_queue[i] >= 100: progress_queue[i] = 100
            
        count = 0
        while progress_queue and progress_queue[0] >= 100:
            count += 1
            progress_queue.popleft()
            speeds_queue.popleft()
        
        if count > 0: answer.append(count)
    return answer